onyo/commands/tests/test_unset.py

```diff
@@ -2,494 +2,383 @@
-    assert f"Field {no_key} does not exist in " in ret.stderr
+    assert "No assets" in ret.stdout
+    assert not ret.stderr
     assert ret.returncode == 0
-    fsck(repo)
+    assert repo.git.is_clean_worktree()
 
 
-@pytest.mark.repo_contents(*contents)
+@pytest.mark.repo_contents(*convert_contents([t for t in asset_contents if "num" in t[1]]))
 def test_unset_multiple_assets(repo: OnyoRepo) -> None:
-    """
-    Test that `onyo unset --keys KEY --path ASSET` removes keys from of assets.
-    """
-    key = list(content_dict.keys())[0]
+    """Test that `onyo unset --keys KEY --path ASSET [ASSET2 ...]` removes keys from of assets."""
+    key = "num"
+    assets = repo.get_asset_paths()
 
     # test unsetting keys for multiple assets:
-    ret = subprocess.run(['onyo', '--yes', 'unset', '--keys', key, '--path', *assets], capture_output=True, text=True)
+    ret = subprocess.run(['onyo', '--yes', 'unset', '--keys', key, '--path', *assets],
+                         capture_output=True, text=True)
 
     # verify output
     assert "The following assets will be changed:" in ret.stdout
-    for asset in [p.relative_to(repo.git.root) for p in repo.asset_paths]:
+    for asset in [a.relative_to(repo.git.root) for a in assets]:
         assert str(asset) in ret.stdout
     assert not ret.stderr
     assert ret.returncode == 0
 
     # verify changes, and the repository clean
-    for asset in repo.asset_paths:
-        assert key not in Path(asset).read_text()
-    fsck(repo)
+    for asset in assets:
+        assert key not in asset.read_text()
+    assert repo.git.is_clean_worktree()
 
 
-non_existing_assets: List[List[str]] = [
+@pytest.mark.repo_contents(*convert_contents([t for t in asset_contents if "num" in t[1]]))
+@pytest.mark.parametrize('no_assets', [
     ["single_non_existing.asset"],
     ["simple/single_non_existing.asset"],
-    [assets[0], "single_non_existing.asset"]]
-
-
-@pytest.mark.repo_files(*assets)
-@pytest.mark.parametrize('no_assets', non_existing_assets)
+    [asset_contents[0][0], "single_non_existing.asset"]])
 def test_unset_error_non_existing_assets(repo: OnyoRepo,
                                          no_assets: list[str]) -> None:
-    """
-    Test that `onyo unset --keys KEY --path ASSET` errors correctly for
-    non-existing assets on root, in directories, or if an invalid asset name is
-    in a list of valid ones.
-    """
-    key = list(content_dict.keys())[0]
+    """Test that `onyo unset --keys KEY --path ASSET` errors correctly for non-existing assets."""
+    key = "num"
     ret = subprocess.run(['onyo', 'unset', '--keys', key, '--path', *no_assets],
                          capture_output=True, text=True)
 
     # verify output
     assert not ret.stdout
-    assert "The following paths are neither an inventory directory nor an asset:" in ret.stderr
+    assert "The following paths" in ret.stderr
```

Comment:
That's a bit strange. What you assert to be in the message is the exact wording of the unspecific part. What seems relevant to me here, is that the paths (or at least one) of `no_assets` shows up in that message. `The following paths` could be in any kind of unrelated error message.